plugins/woocommerce/tests/e2e/specs/smoke-tests/upload-plugin.js

```diff
@@ -13,30 +13,36 @@ const {
 	beforeAll,
 } = require( '@jest/globals' );
 
-const { GITHUB_REPOSITORY, PLUGIN_NAME, GITHUB_TOKEN } = process.env;
+const { GITHUB_REPOSITORY, PLUGIN_NAME, GITHUB_TOKEN, PLUGIN_REPOSITORY } = process.env;
+
+// allows us to upload plugins from different repositories.
+const pluginName = PLUGIN_NAME ? PLUGIN_NAME : 'WooCommerce';
+const repository = PLUGIN_REPOSITORY ? PLUGIN_REPOSITORY : GITHUB_REPOSITORY;
 
 let zipUrl;
 let pluginPath;
 
-utils.describeIf( GITHUB_REPOSITORY )( 'Upload and activate plugin', () => {
-	beforeAll( async () => {
-		zipUrl = await getLatestReleaseZipUrl( GITHUB_REPOSITORY, GITHUB_TOKEN );
-
-		pluginPath = await getRemotePluginZip( zipUrl, GITHUB_TOKEN );
+utils.describeIf( GITHUB_REPOSITORY || PLUGIN_REPOSITORY )(
```

Comment:
```suggestion
utils.describeIf( repository )(
```

Reply:
Thanks for the suggestion. I've made the change